### DISCUSSION TITLE
AuthenticationException on dlink_ds_telnet

### DISCUSSION THREAD

**Discussion by dftagliari** [ktbyers/netmiko#2482] [Q&A]
Hello!
I am using netmiko with dlink_ds_telnet to access the switches and get the fdb table. The script it's ok except with the model "D-Link DES3526". 
With "session_log" parameter I saw there is one more asterisk than the letters of the password.
I have tried the parameter "default_enter" with the values "\r\n", "\n" and "\r", but the problem continues.
Someone can help me?

Follow my script and the session_log:

> def dlinktelnet(ip):
    dlink = {
        'device_type': 'dlink_ds_telnet',
        'ip': ip,
        'username': 'admin',
        'password': '123456',
        'default_enter': '\r\n',
        'session_log': 'netmiko_session.log',
    }
    try:
        conn = ConnectHandler(**dlink)
        cmd = 'disable clipaging'
        conn.send_command(cmd)
        cmd = 'show fdb'
        textocru = '\n' + conn.send_command(cmd)
        conn.disconnect()
        return textocru
    except AuthenticationException:
        erro = 'Erro de autenticação no D-Link via telnet!'
        return erro

>               DES-3526 Fast Ethernet Switch Command Line Interface

                            Firmware: Build 5.00-B28

          Copyright(C) 2000-2004 D-Link Corporation. All rights reserved.

username:admin
**####There is one more asterisk:**
password:*******

Fail!

username:

password:******

Fail!

username:

password:******

Fail!

**Comment by aztec102** [comment]
I'm seeing exactly the same problem.

**Comment by ktbyers** [comment]
The one more asterisk is not significant since we always replace any length password with 6 asterisks (we would not want to log the actual length of your password).

@aztec102 Can you show your Netmiko session_log (obviously change anything that is confidential, but let me know what you changed).

Are you sure your credentials are correct?

**Reply by aztec102** [reply]
@ktbyers I only changed the username
```
[0m[2J[1;1H[2J
D
           
                        DES-3200-10 Fast Ethernet Switch
                             Command Line Interface

                           Firmware: Build 1.82.B006
           Copyright(C) 2010 D-Link Corporation. All rights reserved.

UserName:username

PassWord:**********
Fail!

UserName:

PassWord:*********
Error username or password, RADIUS can not accept NULL string!
Fail!

UserName:

PassWord:*********
Error username or password, RADIUS can not accept NULL string!
Fail!

Blocked unauthorized CLI access!
Maximum number of login attempts reached. Telnet session is terminated.

DES-3200-10:1#[0m

**********
* Logout *
**********
```

**Reply by dima386** [reply]
I recommend rewriting the telnet_login method
and adding something at the beginning of the input line.

https://github.com/ktbyers/netmiko/blob/master/netmiko/cisco_base_connection.py

this input does not work in des3526
self.write_channel(self.username + "\r")
self.write_channel(self.password + "\r")

this input works correctly
**self.write_channel("\x1b" + self.username + "\r")
self.write_channel("\x1b" + self.password + "\r")**
